src/components/costindex/CostIndexTable.tsx

```diff
@@ -0,0 +1,207 @@
+import { get, SubmitHandler, useForm } from 'react-hook-form';
+import { AddEditCostIndex } from '../../types';
+import { yupResolver } from '@hookform/resolvers/yup/dist/yup';
+import * as yup from 'yup';
+import { toast } from '../common/toast/ToastManager';
+import { usePageTitle } from '../../utils/usePageTitle';
+import CostIndexSingleTable from './CostIndexSingleTable';
+
+const T_PATH = 'components.costindex.CostIndexTable';
+
+const CostIndexTable = (): JSX.Element => {
+  const { t: translate } = useTranslation();
+  const t = (label: string) => translate(`${T_PATH}.${label}`);
+  usePageTitle(t('indexValue'));
+
+  const { data, isLoading, isError } = useGetCostIndexesQuery();
+  const [addCostIndex, { isLoading: addIsLoading }] = useAddCostIndexMutation();
+
+  const confirmationDialogTarget = React.useRef(null);
+  const openConfirmationDialogButtonRef = React.useRef(null);
+  const [isOpen, setIsOpen] = React.useState(false);
+  const close = () => setIsOpen(false);
+  const titleId = 'info-dialog-title';
+  const descriptionId = 'info-dialog-content';
+
+  const schema = yup.object({
+    valid_from: yup
+      .string()
+      .trim()
+      .matches(/\d{1,2}\.\d{1,2}\.\d{4}/, t('invalidDate'))
+      .required(t(`required`)),
+    value: yup
+      .string()
+      .trim()
+      .matches(/^\d+([.,]\d{1,2})?$/, t('invalidValue'))
+      .required(t(`required`)),
+  });
+  const { handleSubmit, register, setValue, formState, watch, reset, trigger, getValues } = useForm<AddEditCostIndex>({
+    resolver: yupResolver(schema),
+  });
+
+  const errors = formState.errors;
+  const validFromDate = watch('valid_from');
+  const formId = `cost-index-add-form`;
+
+  const validFrom = getValues('valid_from');
+  const value = getValues('value');
+
+  if (isLoading) {
+    return <LoadingSpinner small />;
+  } else if (isError || !data) {
+    return <>Error terror</>;
+  }
+  const fiDateToISO = (date: string) => moment(date, 'D.M.YYYY', true).format('YYYY-MM-DD');
+  const ISODateToFi = (date: string) => moment(date, 'YYYY-MM-DD').format('D.M.YYYY');
+  const getValidTo = (index: number) => {
+    if (index === 0) {
+      return t('indefinite');
+    }
+    return moment(data[index - 1].valid_from, 'YYYY-MM-DD')
+      .subtract(1, 'day')
+      .format('D.M.YYYY');
+  };
+
+  const onSubmit: SubmitHandler<AddEditCostIndex> = (data, event) => {
+    event?.preventDefault();
+    handleFormSubmit(data);
+  };
+
+  const handleFormSubmit = async (data: AddEditCostIndex) => {
+    data.valid_from = fiDateToISO(data.valid_from);
+    data.value = data.value.replace(',', '.');
+
+    if (!addIsLoading) {
+      try {
+        await addCostIndex({ formData: data })
+          .unwrap()
+          .then(() => {
+            toast.show({ type: 'success', content: t(`${T_PATH}.createdSuccessfully`) });
+            reset();
+          });
+      } catch (err: any) {
+        toast.show({ type: 'error' });
```

Comment:
I tried to save a new cost index with the same date as the previous one and got two identical error toasts just saying a generic error.

We have a middleware error toast already in place, so I think we should show the actual error message (if provided by the API) instead. In this case, the `err` had a descriptive error message, that we could show here for the user.

It might not be the prettiest, but I've tried to tackle the same issue here: https://github.com/City-of-Helsinki/att-sales-ui/blob/develop/src/components/installments/ProjectInstallments.tsx#L180-L199